data/lib/src/datasource_impl/document_datasource_impl.dart

```diff
@@ -56,11 +57,17 @@ class DocumentDataSourceImpl implements DocumentDataSource {
   final LinShareHttpClient _linShareHttpClient;
   final RemoteExceptionThrower _remoteExceptionThrower;
   final LinShareDownloadManager _linShareDownloadManager;
+  final LinShareDatabaseManager _linShareDatabaseManager;
```

Comment:
No, We need to create another datasource for `Document`. 
`LocalDocumentDataSource`. Then you should inject both `DocumentDatasourceImp` and `LocalDocumentDatasource` to Repository.